eggbox_potential_sampler/sampling_data_view/sampling_data_view.py

```diff
@@ -0,0 +1,147 @@
+from chaco.tools.api import PanTool, ZoomTool
+
+from force_wfmanager.ui.review.data_view import BaseDataView
+from force_wfmanager.ui.review.plot import Plot
+from force_wfmanager.ui.review.plot import BasePlot, ChacoPlot
+
+
+class ConvergencePlot(BasePlot):
+    """ This is an example of a more complicated plot that could be
+    contributed by a plugin.
+
+    Here, the running minimum of the y-values is displayed on a line
+    plot.
+
+    """
+
+    #: Simple re-labelling of the underlying :class:`BasePlot` trait.
+    y = Enum(values='_value_names', label='convergence variable')
+
+    #: Extra array to hold our custom plot data
+    _custom_data_array = List()
+
+    #: The View of the plot, showing the menu to choose 'y' and the plot itself
+    view = View(
+        VGroup(
+            HGroup(
+                Item('y'),
+            ),
+            UItem('_plot', editor=ComponentEditor())
+        )
+    )
+
+    def plot_cumulative_line(self):
+        """ Create the Chaco line plot. """
+        plot = ChacoPlot(self._plot_data)
+        line_plot = plot.plot(
+            ('iteration', 'cumulative_minimum'),
+            type='line',
+            name='Line plot',
+            marker='circle',
+            bgcolor='white')[0]
+
+        line_plot.tools.append(PanTool(plot))
+        line_plot.overlays.append(ZoomTool(plot))
+
+        self._plot_index_datasource = line_plot.index
+        plot.set(title=self.title, padding=75, line_width=1)
+        self._axis = line_plot
+
+        return plot
+
+    @on_trait_change('x,y')
+    def _update_plot_data(self):
+        """ Override update_plot_data to include the new ancillary
+        array, and an array containing the index of that array.
+
+        """
+        # A recenter request will be executed here, not by the parent's method
+        self.recenter_required = False
+        super()._update_plot_data()
+
+        self._custom_data_array = []
+
+        if self.y is None:
+            return
+
+        y_index = self.analysis_model.value_names.index(self.y)
+        for ind, _ in enumerate(self._data_arrays[y_index], start=1):
+            self._custom_data_array.append(
+                min(self._data_arrays[y_index][:ind]))
+        self._plot_data.set_data('cumulative_minimum',
+                                 self._custom_data_array)
+        self._plot_data.set_data(
+            'iteration',
+            list(range(1, len(self._custom_data_array) + 1))
+        )
+
+        self.resize_plot()
+
+    def __plot_default(self):
+        """ Set the new default plot. """
+        self._plot = self.plot_cumulative_line()
+        return self._plot
+
+    def __plot_data_default(self):
+        """ Add custom columns to the default plot data. """
+        plot_data = self._get_plot_data_default()
+        plot_data.set_data('cumulative_minimum', [])
+        plot_data.set_data('iteration', [])
+        return plot_data
+
+    def resize_plot(self):
```

Comment:
This doesn't seem to be called when `x` or `y` are changed on the display, since the user has to manually press the "Reset View" button after changing a plot variable to rescale the axes. It could be an issue in `force-wfmanager` actually, since this doesn't seem unique to the example plots in this plugin.

Reply:
Issue raised in force-h2020/force-wfmanager#318